compiler/rustc_ty_utils/src/needs_drop.rs

```diff
@@ -56,8 +55,7 @@ fn has_significant_drop_raw<'tcx>(
         adt_consider_insignificant_dtor(tcx),
         true,
     )
-    .filter(filter_array_elements(tcx, query.param_env))
-    .next()
+    .find(filter_array_elements(tcx, query.param_env))
     .is_some();
```

Comment:
```suggestion
    .any(filter_array_elements(tcx, query.param_env));
```

Reply:
```
error[E0277]: expected a `std::ops::FnMut<(Result<rustc_middle::ty::Ty<'tcx>, AlwaysRequiresDrop>,)>` closure, found `impl for<'a> Fn(&'a Result<rustc_middle::ty::Ty<'_>, AlwaysRequiresDrop>) -> bool`
    --> compiler/rustc_ty_utils/src/needs_drop.rs:58:10
     |
58   |     .any(filter_array_elements(tcx, query.param_env));
     |      --- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected an `FnMut<(Result<rustc_middle::ty::Ty<'tcx>, AlwaysRequiresDrop>,)>` closure, found `impl for<'a> Fn(&'a Result<rustc_middle::ty::Ty<'_>, AlwaysRequiresDrop>) -> bool`
     |      |
     |      required by a bound introduced by this call
     |
     = help: the trait `std::ops::FnMut<(Result<rustc_middle::ty::Ty<'tcx>, AlwaysRequiresDrop>,)>` is not implemented for `impl for<'a> Fn(&'a Result<rustc_middle::ty::Ty<'_>, AlwaysRequiresDrop>) -> bool`
     = note: expected a closure with arguments `(&Result<rustc_middle::ty::Ty<'_>, AlwaysRequiresDrop>,)`
                found a closure with arguments `(Result<rustc_middle::ty::Ty<'tcx>, AlwaysRequiresDrop>,)`
note: required by a bound in `std::iter::Iterator::any`
    --> /home/matthias/vcs/github/rust/library/core/src/iter/traits/iterator.rs:2857:12
     |
2854 |     fn any<F>(&mut self, f: F) -> bool
     |        --- required by a bound in this associated function
...
2857 |         F: FnMut(Self::Item) -> bool,
     |            ^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `Iterator::any`
```

Reply:
You should be able to fix it by introducing a closure

Reply:
Actually, just edit the signature of `filter_array_elements` to take a `Result<Ty<'tcx>, NeedsDrop>` rather than a `&Result<..>`?